app/controllers/api/v1/links_controller.rb

```diff
@@ -12,4 +12,18 @@ def create
       render json: { errors: @link.errors }
     end
   end
+
+  def show
+    param = { long_url: params[:url] }
+    @link = @user.links.build(param)
```

Comment:
Why are you trying to create a new link in the #show action? this should be something like:

```
@link = Link.find_by_short_url(params[:id])
if @link
.....
```